### PR TITLE
migrate boot loader interface doc from wiki into markdown (split out of #10495)

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: docs/BOOT_LOADER_INTERFACE.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  host. The value should be formatted as numeric, NUL-terminated, decimal
+  string, in UTF-16. The time is specified in µs.
+
+* Similar, the EFI variable `LoaderConfigTimeoutOneShot` contains a boot menu
+  time-out for a single following boot. It is set by the OS in order to request
+  display of the boot menu on the following boot. When set overrides
+  `LoaderConfigTimeout`. It is removed automatically after being read by the
+  boot loader, to ensure it only takes effect a single time. This value is
+  formatted the same way as `LoaderConfigTimeout`. If set to `0` the boot menu
+  time-out is turned off, and the menu is shown indefinitely.
```
Comment (keszybz) [inline docs/BOOT_LOADER_INTERFACE.md:35]: So this is not implemented yet, but that should be OK. We just shouldn't release in this state.

File: docs/BOOT_LOADER_INTERFACE.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  be used to let the OS know which boot menu entries are available. A boot
+  loader entry identifier should be a short, non-empty alphanumeric string
+  (possibly containing `-`, too). The list should be in the order the entries
+  are shown on screen during boot. See below regarding a recommended vocabulary
+  for boot loader entry identifiers.
+
+* The EFI variable `LoaderEntryDefault` contains the default boot loader entry
+  to use. It contains a NUL-terminated boot loader entry identifier.
+
+* Similar, the EFI variable `LoaderEntryOneShot` contains the default boot
```
Comment (keszybz) [inline docs/BOOT_LOADER_INTERFACE.md]: "Similarly" (for "contains")

File: docs/BOOT_LOADER_INTERFACE.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  loader entry to use for a single following boot. It is set by the OS in order
+  to request booting into a specific menu entry on the following boot. When set
+  overrides `LoaderEntryDefault`. It is removed automatically after being ready
```
Comment (keszybz) [inline docs/BOOT_LOADER_INTERFACE.md]: "read"

File: docs/BOOT_LOADER_INTERFACE.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  time-out for a single following boot. It is set by the OS in order to request
+  display of the boot menu on the following boot. When set overrides
+  `LoaderConfigTimeout`. It is removed automatically after being read by the
+  boot loader, to ensure it only takes effect a single time. This value is
+  formatted the same way as `LoaderConfigTimeout`. If set to `0` the boot menu
+  time-out is turned off, and the menu is shown indefinitely.
+
+* The EFI variable `LoaderEntries` may contain a series of boot loader entry
+  identifiers, one after the other, each individually NUL terminated. This may
+  be used to let the OS know which boot menu entries are available. A boot
```
Comment (keszybz) [inline docs/BOOT_LOADER_INTERFACE.md]: "are available" → "were discovered by the boot loader" ?

File: docs/BOOT_LOADER_INTERFACE.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+* Similar, the EFI variable `LoaderEntryOneShot` contains the default boot
+  loader entry to use for a single following boot. It is set by the OS in order
+  to request booting into a specific menu entry on the following boot. When set
+  overrides `LoaderEntryDefault`. It is removed automatically after being ready
+  by the boot loader, to ensure it only takes effect a single time. This value
+  is formatted the same way as `LoaderEntryDefault`.
+
+* The EFI variable `LoaderEntrySelected` contains the boot loader entry
+  identifier currently being booted. It is set by the boot loader and read by
```
Comment (keszybz) [inline docs/BOOT_LOADER_INTERFACE.md]: "being booted" → "that was booted" (This makes less assumptions about the what happens after boot.)

File: docs/BOOT_LOADER_INTERFACE.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  the OS in order to identify which entry has been used for the current boot.
+
+* The EFI variable `LoaderFeatures` contains a 64bit unsigned integer with a
+  number of flags bit that are set by the boot loader and passed to the OS and
```
Comment (keszybz) [inline docs/BOOT_LOADER_INTERFACE.md]: "flag bits"

File: docs/BOOT_LOADER_INTERFACE.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  * `1 << 0` → The boot loader honours `LoaderConfigTimeout` when set.
+  * `1 << 1` → The boot loader honours `LoaderConfigTimeoutOneShot` when set.
+  * `1 << 2` → The boot loader honours `LoaderEntryDefault` when set.
+  * `1 << 3` → The boot loader honours `LoaderEntryOneShot` when set.
+  * `1 << 4` → The boot loader supports boot counting as described in [Automatic Boot Assessment](https://systemd.io/AUTOMATIC_BOOT_ASSESSMENT).
+
+If `LoaderTimeInitUSec` and `LoaderTimeExecUSec` are set, `systemd-analyze`
+will include them in its boot-time analysis.  If `LoaderDevicePartUUID` is set,
+systemd will mount the ESP that was used for the boot to `/boot`, but only if
+that directory is empty otherwise, and only if no other file systems are
```
Comment (keszybz) [inline docs/BOOT_LOADER_INTERFACE.md]: "otherwise" here is confusing.

**Comment by poettering** [comment]
I force pushed a new version now. No changes besides the ones @keszybz pointed out. Because these are minor and only in documentation I am taking the liberty to set the green label.

**Pull request merged**